packages/@lwc/ssr-compiler/src/compile-template/transformers/element.ts

```diff
@@ -41,17 +42,32 @@ const bYield = (expr: EsExpression) => b.expressionStatement(b.yieldExpression(e
 // TODO [#4714]: scope token renders as a suffix for literals, but prefix for expressions
 const bConditionalLiveYield = esTemplateWithYield`
     {
-        const shouldRenderScopeToken = ${/* isClass */ is.literal} &&
+        const attrName = ${/* attribute name */ is.literal};
+        let attrValue = ${/* attribute value expression */ is.expression};
+        const isHtmlBooleanAttr = ${/* isHtmlBooleanAttr */ is.literal};
+
+        const shouldRenderScopeToken = attrName === 'class' &&
             (hasScopedStylesheets || hasScopedStaticStylesheets(Cmp));
         const prefix = shouldRenderScopeToken ? stylesheetScopeToken + ' ' : '';
 
-        const attrValue = ${/* attribute value expression */ is.expression};
-        const valueType = typeof attrValue;
+        // Global HTML boolean attributes are specially coerced into booleans
+        // https://github.com/salesforce/lwc/blob/f34a347/packages/%40lwc/template-compiler/src/codegen/index.ts#L450-L454
+        if (isHtmlBooleanAttr) {
+            attrValue = attrValue ? '' : undefined;
+        }
+
+        // Global HTML "tabindex" attribute is specially massaged into a stringified number
+        // This follows the historical behavior in api.ts:
+        // https://github.com/salesforce/lwc/blob/f34a347/packages/%40lwc/engine-core/src/framework/api.ts#L193-L211
+        if (attrName === 'tabindex') {
+            const shouldNormalize = attrValue > 0 && !(attrValue === true || attrValue === false);
+            attrValue = shouldNormalize ? 0 : attrValue;
+        }
 
-        if (attrValue && (valueType === 'string' || valueType === 'boolean')) {
-            yield ' ' + ${/* attribute name */ is.literal};
-            if (valueType === 'string') {
-                yield \`="\${prefix}\${htmlEscape(attrValue, true)}"\`;
+        if (attrValue !== undefined && attrValue !== null) {
+            yield ' ' + attrName;
+            if (attrValue !== '') {
```

Comment:
Follow-up: how does this work for scoped styles combined with `class=""`? The `prefix` should presumably be rendered.

Reply:
https://github.com/salesforce/lwc/pull/4789